src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java

```diff
@@ -2128,6 +2117,30 @@ public Optional<SecureSettingsFactory> getSecureSettingFactory(Settings settings
         return Optional.of(new OpenSearchSecureSettingsFactory(threadPool, sks, sslExceptionHandler, securityRestHandler));
     }
 
+    @SuppressWarnings("removal")
+    private void tryAddSecurityProvider() {
```

Comment:
nit: can we add UT for this?

Reply:
@shikharj05 The refactor retains the same functionality so should be covered by existing tests. What additional test would you propose? Do you think it will hold up the change?

Reply:
If existing tests cover the changes already, it should be okay.